ryven/main/Ryven.py

```diff
@@ -1,105 +1,553 @@
+
+        args.project = pathlib.Path(exampledir, args.example).with_suffix('.json')
+
+    # Make a `set` of nodes
+    args.nodes = set(args.nodes)
+
+    return args, remaining_args
+
+
+def run(*args_,
+        qt_app=None, gui_parent=None, use_sysargs=True,
+        **kwargs):
+    """Start the Ryven window.
+
+    The `*args` and `**kwargs` arguments correspond to their positional and
+    optional command line equivalents, respectively (see `parse_args()`).
+    Optional keyword arguments are specified without the leading double hyphens
+    '--'. As a name, the corresponding 'dest' value of `add_argument` has to
+    be used. E.g. the command line:
+        ryven --window-theme=light --nodes=std --nodes=linalg myproject.json
+    becomes:
+        run('myproject.json', window_theme='light', nodes=['std', 'linalg'])
+
+    Note
+    ----
+    The `*args` and `**kwargs` takes predecence and overwrites the values
+    specified on the command line (or the default values, if
+    `use_sysargs=False` is given). The exception are lists, which are appended,
+    e.g. in the example from above, the two nodes 'std' and 'linalg' are added
+    at the end of the nodes supplied at the command line.
+
+    Parameters
+    ----------
+    qt_app : QApplication, optional
+        The `QApplication` to be used. If `None` a `QApplication` is generated.
+        The default is `None`.
+    gui_parent : QWidget, optional
+        The parent `QWidget`.
+        The default is `None`.
+    use_sysargs : bool, optional
+        Whether the command line arguments should be used.
+        The default is `True`.
+    *args_ : str
+        Corresponding to the positional command line argument(s).
+    **kwargs : any
+        Corresponding to the keyword command line arguments.
+
+    Raises
+    ------
+    TypeError
+        Raised, if keyword argument is not specified by the argument parser or
+        the wrong number of positional arguments are specified.
+
+    Returns
+    -------
+    None|Main Window
+
+
+    """
+
+    #
+    # Process command line and method's arguments
+    #
+
+    if use_sysargs:
+        # Get parsed command line arguments
+        args, remaining_args = parse_args()
+    else:
+        # Get default values
+        args, remaining_args = parse_args(just_defaults=True)
+
+    # Update command line arguments with keyword arguments to run()
+    for key, value in kwargs.items():
+        if not hasattr(args, key):
+            raise TypeError(
+                f"run() got an unexpected keyword argument '{key}'")
+        if isinstance(getattr(args, key), list):
+            getattr(args, key).extend(value)
+        else:
+            setattr(args, key, value)
+
+    # Update the 'project' argument with the positional arguments to run()
+    # Note, this is intentionally generic, so that changes in `parse_args`
+    # does not require changes here!
+    if isinstance(args.project, list):  # Multiple 'project' arguments possible
```

Comment:
I kinda want to remove this. While it might be possible to do this (under constraints), this depends possibly on many other sources, primarily MainWindow, all of which I wrote with the assumption that this is not possible. So I would rather make support for this another tiny PR

Reply:
Well … At the moment it's not possible to specify more than one project file, because the command line parser has
```python3
    parser.add_argument(
        nargs='?',
        dest='project',
        metavar='PROJECT',
        help='the project file or example to be loaded; '
             'use "-" for standard input')
```
which only allows one project file – or none, so you cannot specify more than one project file as argument to `run()`. It might be some overkill, but might reduce errors later, if ryven ever grows the possibility to open more than one project file: Then just the `nargs=?` keyword parameter has to be changed …

Having said that, I do not insist on it.

Reply:
ah ok, then I'll just add a note about this to the code

Reply:
8d0791f678b415c5d6b25ab7978df1a68bcc8d30